freeipa/src/main/java/com/sequenceiq/freeipa/service/stack/FreeIpaScalingValidationService.java

```diff
@@ -0,0 +1,107 @@
+package com.sequenceiq.freeipa.service.stack;
+
+import static java.util.function.Predicate.not;
+
+import java.util.List;
+import java.util.Objects;
+import java.util.Set;
+import java.util.stream.Collectors;
+
+import javax.inject.Inject;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.stereotype.Service;
+
+import com.sequenceiq.cloudbreak.common.exception.BadRequestException;
+import com.sequenceiq.freeipa.api.v1.freeipa.stack.model.common.FormFactor;
+import com.sequenceiq.freeipa.api.v1.freeipa.stack.model.scale.ScalingPath;
+import com.sequenceiq.freeipa.api.v1.operation.model.OperationType;
+import com.sequenceiq.freeipa.configuration.AllowedScalingPaths;
+import com.sequenceiq.freeipa.entity.InstanceMetaData;
+import com.sequenceiq.freeipa.entity.Stack;
+
+@Service
+public class FreeIpaScalingValidationService {
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(FreeIpaScalingValidationService.class);
+
+    @Inject
+    private AllowedScalingPaths allowedScalingPaths;
+
+    public void validateStackForUpscale(Set<InstanceMetaData> allInstances, Stack stack, ScalingPath scalingPath) {
+        if (allInstances.isEmpty()) {
+            throwErrorForNoInstance();
+        } else if (allInstances.size() >= FormFactor.HA.getInstanceCount()) {
+            LOGGER.warn("FreeIPA instance count is bigger then allowed. Size: [{}]", allInstances.size());
+            throw new BadRequestException("Upscaling currently only available for FreeIPA installation with 1 or 2 instances");
+        } else if (isAnyInstanceInNotAvailableState(allInstances)) {
+            throwErrorForNotAvailableInstances(allInstances);
+        } else if (!stack.isAvailable()) {
+            throwErrorForUnavailableStack(stack, OperationType.UPSCALE);
+        } else if (scalingPathDisabled(scalingPath)) {
+            throwErrorForUnsupportedScalingPath(scalingPath, OperationType.UPSCALE);
+        }
```

Comment:
the difference between downscale and upscale validation is the second condition and the operationType used for throwing error, so maybe it is worth to use a common validation method for both validation. also `else if` conditions are not necessary since all validations will throw an exception if any applies